docs/DeveloperGuide.md

```diff
@@ -6,8 +6,248 @@
+
+![Step 3](team/aiman_assignment/step3.png)
+
+__Step 4.__ The `assignHousekeeperCommand` object parses the name of the housekeeper and the ID of the room and send both to the the `assignmentMap`.
+
+![Step 4](team/aiman_assignment/step4.png)
+
+__Step 5.__ The `assignmentMap` looks for the appropriate `Room` object for `301` in the `RoomList` and then looks for the appropriate `Housekeeper` object for `Susan`. It then adds both to the hashmap contained inside itself.
+
+![Step 5](team/aiman_assignment/step5.png)
+
+__Interaction__ To understand the interaction between objects of these classes in order to actually add an entry into the hash map inside an `AssignmentMap`, refer to the sequence diagram below.
+
+![Step Sequence Diagram](team/aiman_assignment/sequence.png)
+
+The command object passes on the ID of the room and the name of the housekeeper to the `AssignmentMap` object after parsing the input appropriately. The `addAssignment` function first locates the appropriate `Room` object from `RoomList`, and then looks for the `Housekeeper` object in the `HousekeeperList` and finally adds the mapping to the hash map.
+
+#### Adding a customer's satisfaction rating
+
+The mechanism for adding a customer satisfaction (represented by the Satisfaction class) is facilitated
+by `AddSatisfactionCommand`. The constructor parses user input to create a `Satisfaction` object (which contains
+attributes storing the customer's name and their satisfaction rating, represented as an integer from 1 to 5).
+`AddSatisfactionCommand` extends `Command` and contains an override of the `Command` class's execute method.
+This execute method adds the `Satisfaction` object created from parsing user command into `satisfactionList`, the
+object of type `SatisfactionList` that stores the `Satisfaction` objects. Additionally, it implements the
+following operations:
+* `AddSatisfactionCommand#extractCustomerName(String userInput)`: Extracts the customer's name (a string) from the
+  provided string of user input.
+* `AddSatisfactionCommand#extractSatisfactionValue(String userInput)`: Extracts the customer's satisfaction rating
+  (an integer from 1 to 5) from the provided string of user input.
+* `AddSatisfactionCommand#getSatisfaction()`: Grabs the value of the `satisfaction` instance variable.
+* `AddSatisfactionCommand#setSatisfaction(Satisfaction satisfaction)`: Sets the `satisfaction` instance variable to the
+  provided `Satisfaction` object.
+
+Given below is an example usage scenario and how the mechanism for adding a satisfaction behaves at each step.
+
+Step 1. The user launches the application. In the `Duke` class, an empty instance of the `SatisfactionList` class,
+called `satisfactionList`, is created.
+
+![Step 1 Object Diagram](team/alinazheng_addsatisfactioncommand_uml/AlinaZheng_AddSatisfaction_Step1ObjectDiagram.png)
+
+
+Step 2. The user types the command `add satisfaction Bob 5`. In the `Duke` class, a `Command` object
```

Comment:
Should `Step#` be bolded as per [above](https://ay2122s2-cs2113-t11-1.github.io/tp/DeveloperGuide.html#assign-a-housekeeper-to-a-room) for consistency?